<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>505</width>
    <height>240</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Image Viewer</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="ImageView" name="imageView">
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="renderHints">
       <set>QPainter::Antialiasing|QPainter::SmoothPixmapTransform|QPainter::TextAntialiasing</set>
      </property>
      <property name="transformationAnchor">
       <enum>QGraphicsView::NoAnchor</enum>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>505</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="actionSaveAsCopy"/>
    <addaction name="actionDelete"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionRotateLeft"/>
    <addaction name="actionRotateRight"/>
    <addaction name="separator"/>
    <addaction name="actionOptions"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <widget class="QMenu" name="menuSort">
     <property name="title">
      <string>Sort</string>
     </property>
     <addaction name="actionSortName"/>
     <addaction name="actionSortDate"/>
     <addaction name="separator"/>
     <addaction name="actionSortAscending"/>
     <addaction name="actionSortDescending"/>
    </widget>
    <addaction name="actionZoomOriginal"/>
    <addaction name="actionZoomFit"/>
    <addaction name="separator"/>
    <addaction name="menuSort"/>
    <addaction name="separator"/>
    <addaction name="actionFullscreen"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <widget class="QMenu" name="menuNavigation">
    <property name="title">
     <string>Navigation</string>
    </property>
    <addaction name="actionPreviousImage"/>
    <addaction name="actionNextImage"/>
    <addaction name="actionFirstImage"/>
    <addaction name="actionLastImage"/>
    <addaction name="separator"/>
    <addaction name="actionSlideshow"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuView"/>
   <addaction name="menuNavigation"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>BottomToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionZoomOriginal"/>
   <addaction name="actionZoomFit"/>
   <addaction name="separator"/>
   <addaction name="actionPreviousImage"/>
   <addaction name="actionSlideshow"/>
   <addaction name="actionNextImage"/>
   <addaction name="separator"/>
   <addaction name="actionRotateLeft"/>
   <addaction name="actionRotateRight"/>
   <addaction name="actionDelete"/>
   <addaction name="separator"/>
   <addaction name="actionFullscreen"/>
  </widget>
  <action name="actionOpen">
   <property name="icon">
    <iconset theme="document-open">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Open...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset theme="document-save">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSaveAsCopy">
   <property name="icon">
    <iconset theme="document-save-as">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Save as copy...</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="icon">
    <iconset theme="application-exit">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionRotateLeft">
   <property name="icon">
    <iconset theme="object-rotate-left">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Rotate Left</string>
   </property>
  </action>
  <action name="actionRotateRight">
   <property name="icon">
    <iconset theme="object-rotate-right">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Rotate Right</string>
   </property>
  </action>
  <action name="actionZoomOriginal">
   <property name="icon">
    <iconset theme="zoom-original">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Zoom 1:1</string>
   </property>
   <property name="shortcut">
    <string>0</string>
   </property>
  </action>
  <action name="actionZoomFit">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="zoom-fit-best">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Zoom Fit</string>
   </property>
   <property name="shortcut">
    <string>5</string>
   </property>
  </action>
  <action name="actionFullscreen">
   <property name="icon">
    <iconset theme="view-fullscreen">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Fullscreen</string>
   </property>
   <property name="shortcut">
    <string>F11</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::WindowShortcut</enum>
   </property>
  </action>
  <action name="actionNextImage">
   <property name="icon">
    <iconset theme="go-next">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Next</string>
   </property>
   <property name="shortcut">
    <string>D</string>
   </property>
  </action>
  <action name="actionPreviousImage">
   <property name="icon">
    <iconset theme="go-previous">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Previous</string>
   </property>
   <property name="shortcut">
    <string>A</string>
   </property>
  </action>
  <action name="actionFirstImage">
   <property name="icon">
    <iconset theme="go-first">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>First</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
  </action>
  <action name="actionLastImage">
   <property name="icon">
    <iconset theme="go-last">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Last</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+D</string>
   </property>
  </action>
  <action name="actionSlideshow">
   <property name="icon">
    <iconset theme="media-playback-start">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Start slideshow</string>
   </property>
   <property name="shortcut">
    <string>Space</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="icon">
    <iconset theme="help-about">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>About...</string>
   </property>
  </action>
  <action name="actionDelete">
   <property name="icon">
    <iconset theme="edit-delete">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Delete...</string>
   </property>
   <property name="shortcut">
    <string>Del</string>
   </property>
  </action>
  <action name="actionSortName">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Name</string>
   </property>
  </action>
  <action name="actionSortDate">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Date</string>
   </property>
  </action>
  <action name="actionSortAscending">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Ascending</string>
   </property>
  </action>
  <action name="actionSortDescending">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Descending</string>
   </property>
  </action>
  <action name="actionOptions">
   <property name="icon">
    <iconset theme="preferences-other">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Options...</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ImageView</class>
   <extends>QGraphicsView</extends>
   <header>Src/MainWindow/ImageView.h</header>
   <slots>
    <signal>fileChanged(QString,QImage)</signal>
    <signal>zoomFitCanceled()</signal>
    <slot>zoomFit(bool)</slot>
    <slot>nextImage()</slot>
    <slot>previousImage()</slot>
    <slot>zoomOriginal()</slot>
    <slot>rotateLeft()</slot>
    <slot>rotateRight()</slot>
    <slot>firstImage()</slot>
    <slot>lastImage()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionAbout</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>about()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>234</x>
     <y>111</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionDelete</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>deleteImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>234</x>
     <y>111</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionFullscreen</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>fullscreen()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>234</x>
     <y>111</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionNextImage</sender>
   <signal>triggered()</signal>
   <receiver>imageView</receiver>
   <slot>nextImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>125</x>
     <y>92</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>openImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>234</x>
     <y>111</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOptions</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>options()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>234</x>
     <y>111</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionPreviousImage</sender>
   <signal>triggered()</signal>
   <receiver>imageView</receiver>
   <slot>previousImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>125</x>
     <y>92</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>234</x>
     <y>111</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRotateLeft</sender>
   <signal>triggered()</signal>
   <receiver>imageView</receiver>
   <slot>rotateLeft()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>125</x>
     <y>92</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSaveAsCopy</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>saveImageAs()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>234</x>
     <y>111</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRotateRight</sender>
   <signal>triggered()</signal>
   <receiver>imageView</receiver>
   <slot>rotateRight()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>125</x>
     <y>92</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSlideshow</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>slideshow()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>234</x>
     <y>111</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSortAscending</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>sortAscending()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>234</x>
     <y>111</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSortDescending</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>sortDescending()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>234</x>
     <y>111</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSortDate</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>sortDate()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>234</x>
     <y>111</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSortName</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>sortName()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>234</x>
     <y>111</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionZoomFit</sender>
   <signal>toggled(bool)</signal>
   <receiver>imageView</receiver>
   <slot>zoomFit(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>125</x>
     <y>92</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionZoomOriginal</sender>
   <signal>triggered()</signal>
   <receiver>imageView</receiver>
   <slot>zoomOriginal()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>125</x>
     <y>92</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>imageView</sender>
   <signal>fileChanged(QString,QImage)</signal>
   <receiver>MainWindow</receiver>
   <slot>fileChanged(QString,QImage)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>125</x>
     <y>92</y>
    </hint>
    <hint type="destinationlabel">
     <x>234</x>
     <y>111</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>saveImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>252</x>
     <y>119</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionFirstImage</sender>
   <signal>triggered()</signal>
   <receiver>imageView</receiver>
   <slot>firstImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>252</x>
     <y>111</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLastImage</sender>
   <signal>triggered()</signal>
   <receiver>imageView</receiver>
   <slot>lastImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>252</x>
     <y>111</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>about()</slot>
  <slot>fullscreen()</slot>
  <slot>options()</slot>
  <slot>sortAscending()</slot>
  <slot>sortDescending()</slot>
  <slot>sortName()</slot>
  <slot>sortDate()</slot>
  <slot>openImage()</slot>
  <slot>slideshow()</slot>
  <slot>saveImageAs()</slot>
  <slot>deleteImage()</slot>
  <slot>fileChanged(QString,QImage)</slot>
  <slot>saveImage()</slot>
 </slots>
</ui>
